File: Algorithms/Arrays/Python/kadane.py
# Max Sub Array Sum

def kadane(nums):
  maxSum = currSum = nums[0]

  for num in nums[1:]:
    currSum = max(num, currSum + num)
    maxSum = max(maxSum, currSum)

  return maxSum


# Test Case:

numsArr = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
print(kadane(numsArr))

File: Algorithms/Arrays/JavaScript/kadane.js
// Max Sub Array Sum

var kadane = function (numsArr) {
  maxSum = currSum = numsArr[0]

  for (let i = 1; i < numsArr.length; i++) {
    currSum = Math.max(numsArr[i], currSum + numsArr[i])
    maxSum = Math.max(maxSum, currSum)
  }

  return maxSum
}

// Test Case:

numsArr = [-2, 1, -3, 4, -1, 2, 1, -5, 4]
console.log(kadane(numsArr))
